docs/decisions/0006-transmit-updates-only.rst

```diff
@@ -0,0 +1,87 @@
+-----------------------------
+
+Status
+======
+
+Draft
+
+Context
+=======
+The transmit_content_metadata jenkins job (along with the underlying integrated_channels.transmit_content_metadata()
+task and export() method) makes an uncomfortably large number of calls to the the enterprise-catalog service’s
+/api/v1/enterprise-catalog/<uuid>/get_content_metadata endpoint with a relatively low number of customers. The resulting
+response payloads are usually quite large which results in the service’s shared Aurora DB often being pinned at 100% CPU
+usage and causing a myriad of alerts and warnings.
+
+At the same time the data being requested by this frequently-running job is often unchanged between runs - content
+metadata tends to change in enterprise-catalog roughly once/day.
+
+Our main goals are as follows:
+
+1. Reduce the overall number of calls required by the Integrated Channels in order to retrieve the content metadata for
+any particular transmission.
+
+2. Prevent the Integrated Channels from making an abundance of requests to the enterprise catalog if there is no update
+needed (ie if nothing in the catalog has changed)
+
+What is the transmit_content_metadata job and underlying export method?
+_______________________________________________________________________
+
+transmit_content_metadata is one of the three core tasks of the Integrated Channels, the other two being
+transmit_learner_data and transmit_subsection_learner_data. Specifically transmit_content_metadata's job is to gather
+course metadata contained within the customer's enterprise catalog and transmit create, update and deletes of course
+entities, through customer configurations, to the external LMS'.
+
+The `export()` method is the main method responsible for fetching said content metadata. It will take all listed
+`catalogs to transmit` under the customer's configurations (otherwise will default to all the customer's catalogs if
+`catalogs to transmit` is not provided) and request content metadata from the enterprise catalog. These catalogs can
+range from "all possible content" to a specific, single course.
+
+Decisions
+=========
+
+In order to address these performance issues and to more effectively handle on-boarding more customers to the Integrated
+Channels, we have decided to address requests to the enterprise catalog on two fronts:
+
+1. Increase the page size of requests going to `/api/v1/enterprise-catalog/<uuid>/get_content_metadata` from a base of
+10 to 100. This will reduce the number of calls necessary to retrieve the entire catalog from 150~ to about 15
+(as of 07/2021).
+
+2. Add an additional request/check to the Integrated Channel's exporter that will retrieve the last time either a
+catalog or the associated content metadata of a catalog has been changed (whichever is most recent). This timestamp will
+be compared to the most recent time the customer has had a successful transmission of that particular catalog. If there
```

Comment:
Yesterday we decided on "the most recent catalog update timestamp that has been successfully exported already" instead of what you have here, correct?

Reply:
yes sorry, forgot to refactor